### PR TITLE
port the cve_2019_6977 fix patch and add test

### PR DESCRIPTION
**Pull request by willson-chen**
port the cve_2019_6977 fix patch from https://gist.github.com/cmb69/1f36d285eb297ed326f5c821d7aafced

### CODE REVIEW

**Comment by willson-chen** [comment]
@vapier @cmb69 hi, could you please spare some time to review this PR.

CVE vuln should be fixed asap. And a new revision is expected after this vuln released.

thx! :)

File: tests/gdimagecolormatch/cve_2019_6977.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Test for CVE-2019-6977
+
```
Comment (cmb69) [inline tests/gdimagecolormatch/cve_2019_6977.c]: Nit: superfluous empty line.

Comment (willson-chen) [reply tests/gdimagecolormatch/cve_2019_6977.c]: fixed

**Comment by willson-chen** [comment]
> Thanks! LGTM. Feel free to commit yourself (I suggest to squash the commits, to avoid having partially broken builds in between, e.g. when later doing `git bisect`).

OK. This is the PR I merge myself. :)

**Pull request merged**